imap_data_access/file_validation.py

```diff
@@ -420,16 +519,13 @@ def __init__(self, filename: str | Path):
             SPICE data filename or file path.
         """
         self.filename = Path(filename)
-        if self.filename.suffix == ".csv":
-            all_suffixes = self.filename.suffixes  # Returns ['.spin', '.csv']
-            self.file_extension = "".join(all_suffixes)  # Returns '.spin.csv'
-        else:
-            self.file_extension = self.filename.suffix
-
-        if self.file_extension not in _SPICE_DIR_MAPPING:
+        self.spice_metadata = SPICEFilePath.extract_filename_components(self.filename)
+        if self.spice_metadata is None:
```

Comment:
Maybe have `extract_filename_components` raise this error instead since it is invalid there too.